staging/src/k8s.io/apiserver/pkg/storage/value/encrypt/kms/google/google.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package google
+
+import (
+	"encoding/base64"
+	"fmt"
+
+	cloudkms "google.golang.org/api/cloudkms/v1"
+	"google.golang.org/api/googleapi"
+)
+
+const defaultGKMSKeyRing = "google-kubernetes"
+
+// GKMSService provides Encrypt and Decrypt methods which allow cryptographic operations
+// using Google Cloud KMS service. It implements kms.Service interface.
+type GKMSService struct {
+	parentName      string
+	cloudkmsService *cloudkms.Service
+}
+
+// NewGoogleKMSService creates a Google KMS connection and returns a GKMSService instance which can encrypt and decrypt data.
+func NewGoogleKMSService(projectID, location, keyRing, cryptoKey string, cloud *cloudkms.Service) (*GKMSService, error) {
```

Comment:
Can you suggest another place for this file then? I was thinking of `cloudproviders/providers/gce`, but this can be used on other clouds as well via the `GOOGLE_APPLICATION_CREDENTIALS` env variable. In that light, it may make sense to keep this somewhere independent of a particular cloud provider.
It would satisfy the KMSService interface, and the KMS transformer (not in this PR, available at sakshamsharma#4) will consume it.